src/utils/get-channel-specific-template.ts

```diff
@@ -0,0 +1,22 @@
+import { logger } from "./logger";
+
+export function getChannelSpecificTemplate(
+  webTemplateAndPath: string,
+  isStrategicAppChannel: boolean,
+  templateMappings: Record<string, string>
+): string {
+  if (isStrategicAppChannel === true) {
+    const templateHasMapping = Object.prototype.hasOwnProperty.call(
```

Comment:
Is there a reason we've written it this way? Are we worried about some of these things being undefined in a way that TS doesn't protect us from? Otherwise, could this become: 
```ts
export function getChannelSpecificTemplate(
  webTemplateAndPath: string,
  isStrategicAppChannel: boolean,
  templateMappings: Record<string, string>,
): string {
  if (!isStrategicAppChannel) {
    return webTemplateAndPath;
  }

  const appTemplate = templateMappings[webTemplateAndPath];
  if (appTemplate === undefined) {
    logger.warn(
      `No '${webTemplateAndPath}' property found in templateMappings. Falling back to web template`,
    );
    return webTemplateAndPath;
  }
  return appTemplate;
}
```

Reply:
Thanks for the suggestion @ethanmills. It's just a stylistic thing that I've carried over from JavaScript. I've updated the code to reflect what you've proposed 👍🏽